examples/message/src/App.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import {withDataProviders} from 'data-provider'
+import {connect} from 'react-redux'
+import {compose} from 'redux'
+import {messageProvider} from './dataProviders'
+
+const Message = ({title, body}) => (
+    <div className='message'>
+        <h3>{title}</h3>
+        <p>{body}</p>
+    </div>
+)
+
+// MessageContainer uses a data-provider with needed === true
+const MessageContainer = compose(
+    connect((state) => ({title: state.title, body: state.body})),
+    withDataProviders(() => [messageProvider(true)]),
+)(Message)
+
+const ToggleableMessageContainer = ({showMessage}) => (
+    <div className='message-container'>
+        {showMessage ? <MessageContainer/> : <span>Click the Show button to show the message</span>}
```

Comment:
Change the message to something which self-explains this example. Such as "Message data is pre-loading, and will finish after 4 seconds" or similar.

Reply:
Done